src/IO.Ably.Shared/MessageEncoders/MessageHandler.cs

```
@@ -113,12 +122,17 @@ public byte[] GetRequestBody(AblyRequest request)
                     request.ChannelOptions);
 
             byte[] result;
-            if (_protocol == Protocol.Json)
+            if (_protocol == Protocol.Json || !Config.MsgPackEnabled)
```

This feels wrong.  Shouldn't the build ensure Protocol is always JSON?  If not, we have a problem in the build and should raise an exception no?

---

This *should* not be needed, I was being extra cautious.

---

I put a check in the constructor and throw an exception, something like

`public MessageHandler(ILogger logger, Protocol protocol)
        {
            if(!Config.MsgPackEnabled && protocol == Protocol.MsgPack)
                throw new AblyException("Invalid Protocol 'MsgPack'. MsgPack is disabled in this build.");
...
}`

---

I went back and looked at why I did this this way. The reason is that not having the !Config.MsgPackEnabled check causes the build to fail with 'Error CS0165: Use of unassigned local variable 'result' (136, 65)'. This is because, although we know that _protocol == Protocol.Json will always be true when MsgPack is not enabled, the compiler does not.
The obvious solution to that is to conditionally build in the Protocol.MsgPack enum, so when MsgPack is not enabled Protocol.MsgPack would not exist. I expect doing that will have some side effects and requires changes elsewhere, but it would make reasoning about the protocol easier. I will investigate.

---

I have done the work of removing Protocol.MsgPack altogether, on another branch, diff here https://github.com/ably/ably-dotnet/compare/remove-msgpack...remove-msgpack-alt?expand=1